manifests/init.pp

```diff
@@ -38,8 +38,17 @@
       class { '::puppet_agent::windows::install': }
     }
     else {
-      class { '::puppet_agent::prepare': } ->
-      class { '::puppet_agent::install': } ->
+
+      if $::operatingsystem == 'SLES' and $::operatingsystemmajrelease == '10' {
+        $_package_file_name = "${puppet_agent::package_name}-${puppet_agent::params::master_agent_version}-1.sles10.${::architecture}.rpm"
```

Comment:
Yes, we could do that, pull this into params as puppet_agent::params::package_file_name which defaults to undef, and then have puppet_agent::prepare::package_file_name and puppet_agent::install::package_file_name default to puppet_agent::params::package_file_name.